cluster/src/main/java/org/apache/iotdb/cluster/query/ClusterPlanExecutor.java

```diff
@@ -634,4 +639,39 @@ public void delete(PartialPath path, long startTime, long endTime, long planInde
       throw new QueryProcessException(e);
     }
   }
+
+  /**
+   * the deleteStorageGroupPlan.getPaths() are all storage group paths, it have been processed in
+   * the {@link org.apache.iotdb.cluster.utils.ClusterUtils#setVersionForSpecialPlan(PhysicalPlan,
+   * long)}
+   */
+  @Override
+  protected boolean deleteStorageGroups(DeleteStorageGroupPlan deleteStorageGroupPlan)
+      throws QueryProcessException {
+    AUDIT_LOGGER.info("delete storage group {}", deleteStorageGroupPlan.getPaths());
+    try {
+      IoTDB.metaManager.deleteStorageGroups(deleteStorageGroupPlan.getPaths());
+    } catch (MetadataException e) {
+      throw new QueryProcessException(e);
+    }
+    return true;
```

Comment:
Because the super.deleteStorageGroup() will use the `deleteStorageGroupPlan.getPaths() `to get allRelatedStorageGroupPath, in this phase, the version of the plan's path will be moved. so when executing the plan, it will be failed due to no version. 
So here we override the `deleteStorageGroups()` method in the `ClusterPlanExecutor.java`

Reply:
Can we guarantee compatibility? It looks like we're not going to throw a `PathNotExistException` like we did before